VTUL/dlp-access
# LIBTD-2370: Admin site metadata editing for an Archive with specified…

… identifier

**JIRA Ticket**: (https://webapps.es.vt.edu/jira/browse/LIBTD-2370) (:star:)

# What does this Pull Request do? (:star:)
This PR enables a site admin user to edit an Archive metadata based on the specific identifier searched. The user can remove or update the value for each single-valued field except that a required field can't be removed. Also the user can add, delete or update values for each multi-valued field (value with all white spaces will be removed from the metadata array). Any metadata changes will be recorded in the History table.

# What's the changes? (:star:)

* Functional stateful components were used in this PR with React hooks introduced to manage the states.
* Added form to search archive with user provided "identifier"
* Added form to edit archive metadata
* Used "updateAt" instead of "modified_date" in collection show page
* Replace unnecessary class components with stateless functional components

# How should this be tested?

* Go to "siteAdmin" page, search for an archive with its identifier 
* With returned search result for an archive's metadata, toggle between "edit" and "view" page
* Edit the metadata, try to perform add/delete/update actions and check if the metadata being updated correctly

# Interested parties
Tag (@yinlinchen ) interested parties

(:star:) Required fields
